Packs/CarbonBlackEnterpriseEDR/Integrations/CarbonBlackEnterpriseEDR/CarbonBlackEnterpriseEDR.py

```diff
@@ -0,0 +1,1316 @@
+    result = client.get_file_device_summary_request(sha256)
+    readable_output = tableToMarkdown('The file device summary', result)
+    results = CommandResults(
+        outputs_prefix='CarbonBlackEEDR.File',
+        outputs_key_field='sha256',
+        outputs=result,
+        readable_output=readable_output,
+        raw_response=result
+    )
+    return results
+
+
+def get_file_metadata_command(client: Client, args: Dict) -> CommandResults:
+    sha256 = args.get('sha256')
+    result = client.get_file_metadata_request(sha256)
+    headers = ['SHA256', 'file_size', 'original_filename', 'internal_name', 'os_type', 'comments']
+    contents = {
+        'SHA256': result.get('sha256'),
+        'file_size': result.get('file_size'),
+        'internal_name': result.get('internal_name'),
+        'original_filename': result.get('original_filename'),
+        'comments': result.get('comments'),
+        'os_type': result.get('os_type')
+    }
+    context = {
+        'sha256': result.get('sha256'),
+        'architecture': result.get('architecture'),
+        'available_file_size': result.get('available_file_size'),
+        'charset_id': result.get('charset_id'),
+        'comments': result.get('comments'),
+        'company_name': result.get('company_name'),
+        'file_available': result.get('file_available'),
+        'file_description': result.get('file_description'),
+        'file_size': result.get('file_size'),
+        'file_version': result.get('file_version'),
+        'internal_name': result.get('internal_name'),
+        'lang_id': result.get('lang_id'),
+        'md5': result.get('md5'),
+        'original_filename': result.get('original_filename'),
+        'os_type': result.get('os_type'),
+        'product_description': result.get('product_description'),
+        'product_name': result.get('product_name'),
+        'product_version': result.get('product_version')
+    }
+
+    readable_output = tableToMarkdown('The file metadata', contents, headers, removeNull=True)
+    results = CommandResults(
+        outputs_prefix='CarbonBlackEEDR.File',
+        outputs_key_field='sha256',
+        outputs=context,
+        readable_output=readable_output,
+        raw_response=result
+    )
+    return results
+
+
+def get_file_command(client: Client, args: Dict) -> CommandResults:
+    sha256 = argToList(args.get('sha256'))
+    expiration_seconds = args.get('expiration_seconds')
+    download_to_xsoar = args.get('download_to_xsoar')
+
+    result = client.get_file_request(sha256, expiration_seconds)
+    contents = []
+
+    found_files = result.get('found', [])
+    for file_ in found_files:
+        contents.append({
+            'sha256': file_.get('sha256'),
+            'url': f"[{file_.get('url')}]({file_.get('url')})"
+        })
+
+        if download_to_xsoar == 'true':
+            request = requests.get(file_.get('url'))
+            demisto.results(fileResult('result.zip', request.content))
```

Comment:
the `zip` extension here is problematic, for example `87976f3430cc99bc939e0694247c0759961a49832b87218f4313d6fc0bc3a776` is hash of an `exe` file

lets change the logic here to first get the metadata, and try to see extract the file extension for the filename, for example `RuntimeBroker.exe` for the sha mentioned above, if it exist, and use `.exe` as default as i see most files are as such

also, need to change the filename to download to be the sha256 and not `result`